src/olympia/bandwagon/views.py

```diff
@@ -115,8 +116,21 @@ class CollectionAddonViewSet(ModelViewSet):
     # This endpoint can be quite slow so it's cached for one hour for all
     # anonymous users.
     @method_decorator(cache_page_if_anonymous(60 * 60 * 1))
-    def list(self, *args, **kwargs):
-        return super().list(*args, **kwargs)
+    def list(self, request, *args, **kwargs):
+        # Swap DRF's Response with a HttpResponse so that it's smaller in size
+        # when pickling (data attribute is lost, we only keep the rendered
+        # content), which matters because by default, memcached doesn't
+        # accept values over 1 MB.
+        # Note that this needs to happen before cache_page_if_anonymous() has
+        # seen the response, because it attaches the callback that caches the
+        # response to it. That's why we manually call finalize_response() and
+        # render() here.
+        response = super().list(*args, *kwargs)
```

Comment:
it's a shame we can't do this in `cache_page_if_anonymous` so it could be reused for other endpoints without having to copy this chunk of code into each one.

Reply:
Yeah. Alternatively we could write our own, but this one is fully battle-tested so... We might end up having a base mixin for all API classes to handle this, your optimizations to the translations, etc.